test/main.test.js

```diff
@@ -0,0 +1,495 @@
+const supertest = require('supertest');
+
+function createTestServerFor(router) {
+    const app = express();
+    app.use(router);
+    return new Promise(resolve => {
+        const server = app.listen(() => {
+            resolve({
+                server,
+                port: server.address().port,
+            });
+        });
+    });
+}
+
+const singleFeed = [
+    {
+        id: 'c645cf572a8f5acf8716e4846b408d3b1ca45c58',
+        source: '"use strict";module.exports.world=function(){return"world"};',
+        deps: {},
+        file: './assets/js/bar.js',
+    },
+];
+
+describe('Router class', () => {
+    test('new Router() with no arguments', () => {
+        expect.assertions(1);
+        const router = new Router();
+        expect(router.sink).toBeInstanceOf(SinkMem);
+    });
+
+    test('new Router(sink) with sink argument', () => {
+        expect.assertions(1);
+        const sink = new SinkFs({ path: '/' });
+        const router = new Router(sink);
+        expect(router.sink).toBeInstanceOf(SinkFs);
+    });
+
+    test('new Router().router property', () => {
+        expect.assertions(1);
+        const router = new Router();
+        expect(router.router.name).toEqual(express.Router().name); // eslint-disable-line
+    });
+});
+
+describe('Router instance methods', () => {
+    test('postFeedResponseCallback', done => {
+        expect.assertions(6);
+        const router = new Router();
+        const json = jest.fn();
+        const req = {
+            method: 'POST',
+            path: '/feed',
+        };
+        const res = {
+            json,
+            status(responseStatus) {
+                expect(responseStatus).toBe(200);
+                return this;
+            },
+            locals: { response: { file: 'foo' }, track: 'bar' },
+        };
+        router.on('request success', (track, method, path, file) => {
+            expect(track).toBe('bar');
+            expect(method).toBe('POST');
+            expect(path).toBe('/feed');
+            expect(file).toBe('foo');
+            expect(json).toHaveBeenCalledWith({ file: 'foo' });
+            done();
+        });
+
+        router.postFeedResponseCallback()(req, res);
+    });
+
+    test('getTestFileCallback: secure', () => {
+        expect.assertions(2);
+        const router = new Router();
+        const send = jest.fn();
+        const req = {
+            params: { file: 'bar' },
+            headers: { host: 'foo' },
+            secure: true,
+        };
+        const res = {
+            send,
+            status(responseStatus) {
+                expect(responseStatus).toBe(200);
+                return this;
+            },
+        };
+
+        router.getTestFileCallback()(req, res);
+        expect(send.mock.calls[0][0]).toMatchSnapshot();
+    });
+
+    test('getTestFileCallback: insecure', () => {
+        expect.assertions(2);
+        const router = new Router();
+        const send = jest.fn();
+        const req = {
```

Comment:
I'm more fan of using `supertest` than creating fake `req` and `res` objects. Might just be me, though

Reply:
We should do something consistent, I tend to do random stuff :S

Reply:
I've done a little isolation of callback functions (possibly wrong abstraction) in places to avoid more complex integration tests that try to cover every case. To do such isolation I needed to mock the `req` and `res`. Not a big fan either but wanted to just get it all under test. Now that I've done so I might try to refactor a little